x-pack/plugins/elastic_assistant/server/ai_assistant_data_clients/knowledge_base/index.ts

```diff
@@ -169,30 +177,83 @@ export class AIAssistantKnowledgeBaseDataClient extends AIAssistantDataClient {
     this.options.logger.debug(`Checking if ELSER model '${elserId}' is deployed...`);
 
     try {
-      const esClient = await this.options.elasticsearchClientPromise;
-      const getResponse = await esClient.ml.getTrainedModelsStats({
-        model_id: elserId,
-      });
+      if (this.isV2KnowledgeBaseEnabled) {
+        return await this.isInferenceEndpointExists();
+      } else {
+        const esClient = await this.options.elasticsearchClientPromise;
+        const getResponse = await esClient.ml.getTrainedModelsStats({
+          model_id: elserId,
+        });
 
-      // For standardized way of checking deployment status see: https://github.com/elastic/elasticsearch/issues/106986
-      const isReadyESS = (stats: MlTrainedModelStats) =>
-        stats.deployment_stats?.state === 'started' &&
-        stats.deployment_stats?.allocation_status.state === 'fully_allocated';
+        // For standardized way of checking deployment status see: https://github.com/elastic/elasticsearch/issues/106986
+        const isReadyESS = (stats: MlTrainedModelStats) =>
+          stats.deployment_stats?.state === 'started' &&
+          stats.deployment_stats?.allocation_status.state === 'fully_allocated';
 
-      const isReadyServerless = (stats: MlTrainedModelStats) =>
-        (stats.deployment_stats?.nodes as unknown as MlTrainedModelDeploymentNodesStats[]).some(
-          (node) => node.routing_state.routing_state === 'started'
-        );
+        const isReadyServerless = (stats: MlTrainedModelStats) =>
+          (stats.deployment_stats?.nodes as unknown as MlTrainedModelDeploymentNodesStats[])?.some(
```

Comment:
No changes requested in this pr, however this check doesn't include the model id used by the inference endpoint, which means it's possible for this check to return true, even though there are no longer any nodes capable of performing inference for the assistant.